src/qpx_mmc.cpp

```diff
@@ -0,0 +1,4244 @@
+        // 				drive->mmc=5;
+        drive->rd_capabilities |= DEVICE_DVD_PLUS_R_DL;
+        if (drive->rd_buf[12] & 0x01)
+          drive->wr_capabilities |= DEVICE_DVD_PLUS_R_DL;
+      }
+      get_configuration(drive, FEATURE_DVD_PLUS_RW_DOUBLE_LAYER, &len, NULL);
+      if (len >= 0x0C) {
+        // 				drive->mmc=5;
+        drive->rd_capabilities |= DEVICE_DVD_PLUS_RW_DL;
+        if (drive->rd_buf[12] & 0x01)
+          drive->wr_capabilities |= DEVICE_DVD_PLUS_RW_DL;
+      }
+    }
+    if (drive->wr_capabilities & DEVICE_DVD_RW)
+      drive->rd_capabilities |= DEVICE_DVD_RW;
+    if (drive->wr_capabilities & DEVICE_DVD_R_DL)
+      drive->rd_capabilities |= DEVICE_DVD_R_DL;
+    get_configuration(drive, FEATURE_BD_READ, &len, NULL);
+    if ((len >= 0x08) && (drive->rd_buf[11] >= 28)) {
+      if (drive->rd_buf[16] | drive->rd_buf[17] | drive->rd_buf[18] |
+          drive->rd_buf[19] | drive->rd_buf[20] | drive->rd_buf[21] |
+          drive->rd_buf[22] | drive->rd_buf[23])
+        drive->rd_capabilities |= (DEVICE_BD_RE);
+      if (drive->rd_buf[24] | drive->rd_buf[25] | drive->rd_buf[26] |
+          drive->rd_buf[27] | drive->rd_buf[28] | drive->rd_buf[29] |
+          drive->rd_buf[30] | drive->rd_buf[31])
+        drive->rd_capabilities |= (DEVICE_BD_R);
+      if (drive->rd_buf[32] | drive->rd_buf[33] | drive->rd_buf[34] |
+          drive->rd_buf[35] | drive->rd_buf[36] | drive->rd_buf[37] |
+          drive->rd_buf[38] | drive->rd_buf[39])
+        drive->rd_capabilities |= (DEVICE_BD_ROM);
+    }
+    get_configuration(drive, FEATURE_BD_WRITE, &len, NULL);
+    if ((len >= 0x08) && (drive->rd_buf[11] >= 20)) {
+      if (drive->rd_buf[16] | drive->rd_buf[17] | drive->rd_buf[18] |
+          drive->rd_buf[19] | drive->rd_buf[20] | drive->rd_buf[21] |
+          drive->rd_buf[22] | drive->rd_buf[23])
+        drive->wr_capabilities |= (DEVICE_BD_RE);
+      if (drive->rd_buf[24] | drive->rd_buf[25] | drive->rd_buf[26] |
+          drive->rd_buf[27] | drive->rd_buf[28] | drive->rd_buf[29] |
+          drive->rd_buf[30] | drive->rd_buf[31])
+        drive->wr_capabilities |= (DEVICE_BD_R);
+    }
+    get_configuration(drive, FEATURE_HDDVD_READ, &len, NULL);
+    if ((len >= 0x08)) {
+      drive->rd_capabilities |= DEVICE_HDDVD_ROM;
+      if ((len >= 0x0C)) {
+        if (drive->rd_buf[12] & 0x01)
+          drive->rd_capabilities |= DEVICE_HDDVD_R;
+        if (drive->rd_buf[14] & 0x01)
+          drive->rd_capabilities |= DEVICE_HDDVD_RAM;
+      }
+    }
+    get_configuration(drive, FEATURE_HDDVD_WRITE, &len, NULL);
+    if ((len >= 0x0C)) {
+      if (drive->rd_buf[12] & 0x01)
+        drive->wr_capabilities |= DEVICE_HDDVD_R;
+      if (drive->rd_buf[14] & 0x01)
+        drive->wr_capabilities |= DEVICE_HDDVD_RAM;
+    }
+    if (!drive->silent)
+      printf("** Device is MMC-%d\n", drive->mmc);
+  }
+}
+
+int read_atip(drive_info *drive) {
+  unsigned char data[4];
+  int size = 0;
+  int i;
```

Comment:
Codacy found an issue: [The scope of the variable 'i' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)